src/main/app/case/C100CaseApi.ts

```diff
@@ -11,10 +11,21 @@ import { getServiceAuthToken } from '../auth/service/get-service-auth-token';
 import { UserDetails } from '../controller/AppRequest';
 
 import { Case } from './case';
-import { C100 } from './definition';
+import { C100, State } from './definition';
 export class CaseApi {
   constructor(private readonly axios: AxiosInstance, private readonly logger: LoggerInstance) {}
 
+  public async retrieveCase(): Promise<RetreiveDraftCase[]> {
+    try {
+      const url: string = config.get('services.cos.url') + '/cases';
+      const response = await this.axios.get<RetreiveDraftCase[]>(url);
+      return response.data.filter(caseData => caseData.state === 'AWAITING_SUBMISSION_TO_HMCTS');
```

Comment:
@SangeethaPonnusamy Should we add checks if response is null or undefined ?

Reply:
I have added that check in getController based on the length